src/Authentication/Authentication/custom/common/Permissions.ps1

```diff
@@ -64,33 +87,57 @@ function Permissions_GetOauthData {
         }
         
         $entry = [ordered] @{
-            
             "Id" = $oauth2grant.id
-            "PermissionType" = $oauth2grant.type
+            "PermissionType" = 'Delegated'
+            "Consent" = $oauth2grant.type
             "Name" = $oauth2grant.value
             "Description" = $description
-        
         }
-
-        [PSCustomObject] $entry
+        $permissions = [PSCustomObject] $entry
+        $permissions.PSTypeNames.Insert(0, 'Microsoft.Graph.Custom.Permission')
+        $permissions
 
     }
 }
 
 function Permissions_GetAppRolesData {
+    param (
+        [bool] $online
+    )
     
-    $permissions = Permissions_GetPermissionsData
-    $msAppRoles = $permissions.appRoles 
+    if ($online){
+
+        $permissions = Permissions_GetPermissionsData $online
+        $msAppRoles = $permissions.appRoles
+
+    } else {
+
+        $permissions = Permissions_GetPermissionsData
+        $msAppRoles = $permissions.appRoles
+
+    } 
     
-    ForEach ($oauth2grant in $msAppRoles) {
+    ForEach ($approle in $msAppRoles) {
 
-        $entry = [ordered] @{
-            "Id" = $oauth2grant.id
-            "PermissionType" = $oauth2grant.origin
-            "Name" = $oauth2grant.value
-            "Description" = $oauth2grant.description
+        $consent = If ($approle.origin -eq "Application") { 
+        
+            "Admin"
```

Comment:
Let's use "Administrator," not "Admin. Also, no need for empty lines before and after.

Reply:
I have gotten rid of the empty lines. I left it as admin so that it can match the consent values for the oauth2

Reply:
I agree with your take on `admin` -- our documentation actually uses the term `Admin` for this exact instance of data, and as you point out that is also consistent with the REST API for `oauth2PermissionGrant`. Also, since this is PowerShell, it straddles the world of Presentation UX + Automation / API, so it's ok for it to make a choice that's consistent with an API but not the same as a more "official" presentation that would be found in Azure Portal, which does not support automation.

[Our docs do indeed use admin instead of administrator](https://docs.microsoft.com/en-us/graph/permissions-reference#access-reviews-permissions)